cpp/src/Ice/RequestHandlerCache.cpp

```diff
@@ -324,5 +324,8 @@ RequestHandlerCache::handleException(
         {
             throw; // Retry could break at-most-once semantics, don't retry.
         }
+
+        // gcc complains without this return statement.
```

Comment:
In other sports we add `assert(false)` to make it clear this should never be reached.

Reply:
fixed